Finished 'Builtin and Custom Objects' homework #2

daktest opened the pull request:
- wrote a Television class with notes on how to use it

[#151556921]

daktest committed:
Finished 'Builtin and Custom Objects' homework #2

 - wrote a Television class with notes on how to use it

[#151556921]

wb97531 commented on objects/television.rb:
```diff
@@ -0,0 +1,120 @@
+  VOLUME_RANGE = 0..25
+
+  def initialize
+    @is_on = false
```
conventionally speaking, a '?' should be used for when a method name or instance variable returns a boolean

wb97531 commented:
well done, to clean it up a bit, look for some opportunities to use some attr_readers

daktest commented:
I don't see a use for attr_reader here.  I had originally used :channel and :volume but that allowed a user to retrieve the channel and volume when the Television was not turned on which was behaviour I didn't want.

daktest committed:
Refactored

 - changed `@is_on` to `@powered'
 - added a `on?` method to return `@powered`

I do not see a use for attr_reader here.  I originally had :channel
and :volume but that allowed a user to get the current channel and
volume when the Television was not turned on and that was behavior
I did not want.

[#151556921]

wb97531 commented:
does 'on?' mean something different then 'powered', why are two methods necessary?

so for example: you could just have an attr_reader for on? and get rid of @powered and on? method

daktest commented:
There aren't two methods; @powered is just a variable; on? is a method.

How do you create an attr_reader with a '?' in it?  I tried and got an error.  Am I missing something?

daktest committed:
Refactored some more

 - removed `@powered` and method `on?'
 - added attr_reader :on
 - added alias_method :on?, :on

This is the only way I could get an attr_reader functionality with a '?'.

[#151556921]

jasonnoble commented on objects/television.rb:
```diff
@@ -0,0 +1,122 @@
+#! /usr/bin/env ruby
```
No space between `#!` and `/usr/bin...`

daktest replied:
Gah, I keep doing that!  Don't know why, I don't do it in Perl!

jasonnoble commented on objects/television.rb:
```diff
@@ -0,0 +1,122 @@
+  end
+
+  def set_channel(number)
+    if on?
```
You may want to change this to a logic gate...

    def set_channel(number)
      return unless on?
      ...
   end

jasonnoble commented on objects/television.rb:
```diff
@@ -0,0 +1,122 @@
+
+  def mute
+    if on?
+      if @volume == 0
```
`if muted`

daktest committed:
Refactored again

 - made suggested changes

[#151556921]
